### ISSUE TITLE
functions don't timeout itself if the device doesnt support eSSP.

### ISSUE THREAD

**Issue by ladiko** [pc-coholic/eSSP#7]
i try your code to check via udev.rules if a newly device is capable of "speaking" cctalk or eSSP. If you use the old NV9 (non-USB) with IF17-USB-SSP adapters, they just identify as an FTDI device and i have other devices with the same chipset too. So I have to identify them by "talking to them". While the cctalk function from https://github.com/Baldanos/ccTools throw an error when the device is not able to reply correctly, the eSSP function just hangs. So what I do is to call the python script with

`DEVNAME=${ttyDevice} timeout -s 9 1 /path/to/ssp_test.sh`

the content:

```
#!/usr/bin/python

import os
import sys
import time
import eSSP

try:
	tty = os.environ['DEVNAME']
except:
	tty = '/dev/ttyUSB0'

k = eSSP.eSSP(tty)
try:
	k.setup_request();
except:
	sys.exit(-1)
else:
	sys.exit(0)
```

maybe you have a solution which makes the script "exits itself" ?

**Comment by pc-coholic** [comment]
To be honest, fixing this would probably include a major rewrite... Not saying, that it should not be done - but that's probably nothing, that can be done easily and quickly...

Here is an idea though: Perhaps try to call a simple function, that does not try to process the (expected) reply from the validator.

One of those might be for example sync()

In theory, this should give you one of the following return-values:
0xf0   OK
0xf2   Command not known
0xf3   Wrong number of parameters
0xf4   Parameter out of range
0xf5   Command cannot be processed
0xf6   Software Error
0xf8   FAIL
0xFA   Key not set

By default, the return-value should be 0xF8 (FAIL) - even when there is no data read back from the device...

If that is not working either, please try turning on the debug-logger and post the log-file... Perhaps there is another way around this.

**Comment by pc-coholic** [comment]
Hello!

This should be fixed with #9 - so I'm closing this issue... If you feel like this is not fixing the issue, please let me know and reopen the issue :)
